### DISCUSSION TITLE
How to play audio and video in the theia?

### DISCUSSION THREAD

**Discussion by lxn179208** [eclipse-theia/theia#9285] [Q&A]
### Description
 
I want to choose  playing audio and video when I right click a mp3/mp4 file.
How?

**Comment by vince-fugnitto** [comment]
> I want to choose playing audio and video when I right click a mp3/mp4 file.

@lxn179208 you have two main options:
1. create a custom theia extension (provides more flexibility)
2. create a custom vscode extension (provided there are none already that handle your use-case) - (more limited than a option 1)

The idea would be to create an [`OpenHandler`](https://github.com/eclipse-theia/theia/blob/e945f2efc1b948f2f6f671f047995caac829682f/packages/core/src/browser/opener-service.ts#L28-L55) for `.mp3/.mp4` files, and when opened the file can be played using some library. You can take a look at the source code for other examples of implementations of `OpenHandler`.

**Comment by akosyakov** [comment]
I am not sure it can be done with VS Code extensions, i.e. they don't have access to browser API to play sounds (maybe with webviews) and can we override right click on the file?

**Comment by akosyakov** [comment]
Not sure what you mean by right-click? From context menu?

**Comment by akosyakov** [comment]
Browser API to play sounds: https://developer.mozilla.org/en-US/docs/Web/API/Web_Audio_API

**Comment by lxn179208** [comment]
@akosyakov Right click means "the menu shows when right click the file in the exporer tab".
I hava an idea, add a button jump to the new native tab of the browser.
![image](https://user-images.githubusercontent.com/811605/75626596-b0421700-5c03-11ea-93ac-f3e6132ce855.png)
